src/Plugin/Autoloader.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace Kirby\Plugin;
+
+/**
+ * Autoloader 
+ */
+
+use Closure;
+use Kirby\Filesystem\Dir;
+use Kirby\Toolkit\A;
+use Kirby\Data\Data;
+use Kirby\Exception\Exception;
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Filesystem\F;
+use Kirby\Toolkit\Str;
+use Throwable;
+
+class Autoloader
+{
+
+    public array $extends = [];
+
+    /**
+     * @param string $name Plugin name
+     * @param string $root The root path of the plugin
+     * @param array $data Predefined extension data to extend
+     * @return void|$this 
+     */
+    public function __construct(
+        public string $name,
+        public string $root,
+        public array $data = []
+    ) {
+
+        //Load classes before everything else
+        $classfolder = $root . '/classes/';
+        if (Dir::exists($classfolder)) {
+            $this->_classes($classfolder);
+        }
+
+        foreach (Dir::dirs($this->root) as $dir) {
+            if (method_exists($this, $dir)) {
+                $this->$dir($this->root . '/' . $dir . '/');
+            }
+        }
+    }
+
+    /**
+     * Walk throuh the given path
+     * @param string $root Absolute path to walk
+     * @param Closure $fnc Callback for each file
+     * @return void 
+     */
+    public function _dirWalker(string $root, Closure $fnc)
+    {
+
+        foreach (Dir::index($root, true) as $path) {
+
+            //Check if the path is active
+            if (Str::contains($path, '/_')) {
+                continue;
+            }
+
+            $file = $root . $path;
+            if (F::exists($file)) {
+                $dirname = F::dirname($path);
+                $dirname = $dirname === '.' ? '' : $dirname . '/';
+                $fnc($dirname . F::name($path), $file);
+            }
+        }
+    }
```

Comment:
I think a less flexible and more explicit approach would be better. Reading the code, the `_dirWalker` method comes across quite cryptic and hard to understand. I also don't understand why many extensions would need to run the index vs. just `Dir::files()`. That's why IMO it would be better to leave it to each dedicated `load*()` method to use the appropriate code and not this meta method.

Reply:
That was my initial plan. However, this makes the entire file very repetitive. I use index so that I can include all subfolders at once.